src/ipc/ipc3/handler.c

```diff
@@ -705,6 +705,11 @@ static int ipc_pm_gate(uint32_t header)
 		pm_runtime_disable(PM_RUNTIME_DSP, PLATFORM_PRIMARY_CORE_ID);
 		cpu_restore_secondary_cores();
 	} else {
+		/* before we enable pm runtime and perform D0->D0ix flow
+		 * (primary core powers off secondary cores in ) we have to
+		 * writeback all secondary core data.
+		 */
+		cpu_writeback_secondary_cores();
```

Comment:
We have flows where the host powers off secondary DSP cores using `SOF_IPC_PM_CORE_ENABLE`. Is `SOF_IPC_PM_GATE` also supposed to be able to switch secondary cores off without `SOF_IPC_PM_CORE_ENABLE` being used? As far as I understand in audio context, `SOF_IPC_PM_GATE` would typically be sent if the only active stream is keyword detection, running on the primary core. In that case all other cores should already be powered off?

Reply:
If LPS is involved then all secondary cores will be put to sleep by FW independently.
Check issue #4382 for more info about such scenario